File: android_webview/java/src/org/chromium/android_webview/AwBackForwardCacheSettings.java
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.android_webview;

import org.chromium.build.annotations.NullMarked;
import org.jni_zero.CalledByNative;
import org.jni_zero.JNINamespace;

/** Settings for toggling back/forward cache behavior. */
@JNINamespace("android_webview")
@NullMarked
public class AwBackForwardCacheSettings extends AwSupportLibIsomorphic {
    private final int mTimeoutInSeconds;
    private final int mMaxPagesInCache;

    public AwBackForwardCacheSettings(int timeoutInSeconds, int maxPagesInCache) {
        mTimeoutInSeconds = timeoutInSeconds;
        mMaxPagesInCache = maxPagesInCache;
    }

    @CalledByNative
    public int getTimeoutInSeconds() {
        return mTimeoutInSeconds;
    }

    @CalledByNative
    public int getMaxPagesInCache() {
        return mMaxPagesInCache;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AwBackForwardCacheSettings)) {
            return false;
        }
        AwBackForwardCacheSettings that = (AwBackForwardCacheSettings) o;
        return mTimeoutInSeconds == that.mTimeoutInSeconds
                && mMaxPagesInCache == that.mMaxPagesInCache;
    }
}

File: android_webview/browser/aw_back_forward_cache_settings.cc
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "android_webview/browser/aw_back_forward_cache_settings.h"

#include <memory>

// Must come after all headers that specialize FromJniType() / ToJniType().
#include "android_webview/browser_jni_headers/AwBackForwardCacheSettings_jni.h"

namespace android_webview {

AwBackForwardCacheSettings
AwBackForwardCacheSettings::FromJavaAwBackForwardCacheSettings(
    JNIEnv* env,
    const base::android::JavaRef<jobject>& java_back_forward_cache_settings) {
  return AwBackForwardCacheSettings(
      Java_AwBackForwardCacheSettings_getTimeoutInSeconds(
          env, java_back_forward_cache_settings),
      Java_AwBackForwardCacheSettings_getMaxPagesInCache(
          env, java_back_forward_cache_settings));
}

AwBackForwardCacheSettings::AwBackForwardCacheSettings(int timeout_in_seconds,
                                                       int max_pages_in_cache)
    : timeout_in_seconds_(timeout_in_seconds),
      max_pages_in_cache_(max_pages_in_cache) {}

}  // namespace android_webview

DEFINE_JNI(AwBackForwardCacheSettings)
